src/test/scala/mesosphere/marathon/api/v2/validation/AppValidationTest.scala

```diff
@@ -277,4 +277,65 @@ class AppValidationTest extends UnitTest with ValidationTestLike {
     }
 
   }
+
+  "health check validation" when {
+    val allowedProtocols = Set(
+      AppHealthCheckProtocol.MesosHttp,
+      AppHealthCheckProtocol.MesosHttps,
+      AppHealthCheckProtocol.MesosTcp)
+
+    "is docker app" should {
+      val dockerApp = App(
+        id = "/foo",
+        container = Some(Container(
+          `type` = EngineType.Docker,
+          docker = Some(DockerContainer(
+            image = "xyz")))))
+
+      "pass validation when Mesos HTTP/HTTPS/TCP health Check for ip protocol" in {
+        allowedProtocols.foreach { protocol =>
+          val dockerAppWithMesosHttpHealthCheck = dockerApp.copy(healthChecks =
+            Set(AppHealthCheck(
+              protocol = protocol,
+              port = Some(80),
+              ipProtocol = Some(IpProtocol.Ipv6))))
+
+          basicValidator(dockerAppWithMesosHttpHealthCheck) should be(aSuccess)
+        }
+      }
+
+      "fail for Marathon HTTP health check with ip protocol" in {
+        val dockerAppWithMesosHttpHealthCheck = dockerApp.copy(healthChecks =
+          Set(AppHealthCheck(
+            protocol = AppHealthCheckProtocol.Http,
+            port = Some(80),
+            ipProtocol = Some(IpProtocol.Ipv6))))
+
+        basicValidator(dockerAppWithMesosHttpHealthCheck) should haveViolations(
+          "/healthChecks(0)" -> AppValidationMessages.HealthCheckIpProtocolLimitation)
+      }
+    }
+
+    "is UCR app" should {
+      val ucrApp = App(
+        id = "/foo",
+        container = Some(Container(
+          `type` = EngineType.Mesos,
+          docker = Some(DockerContainer(
+            image = "xyz")))))
+
+      "should fail even for otherwise supported types" in {
+        allowedProtocols.foreach { protocol =>
+          val dockerAppWithMesosHttpHealthCheck = ucrApp.copy(healthChecks =
+            Set(AppHealthCheck(
+              protocol = protocol,
+              port = Some(80),
+              ipProtocol = Some(IpProtocol.Ipv6))))
+
+          basicValidator(dockerAppWithMesosHttpHealthCheck) should haveViolations(
+            "/healthChecks(0)" -> AppValidationMessages.HealthCheckIpProtocolLimitation)
+        }
+      }
+    }
+  }
```

Comment:
One more test without setting the protocol?

Reply:
you mean IP protocol? Hopefully - added...